docs/enhancements/agent-late-binding.md

```diff
@@ -96,23 +97,23 @@ It would be beneficial to move the discovery image management to a separate serv
+- Currently, a cluster ID is baked into the ISO, and the agent relays that to the service upon registration.  With late binding, the InfraEnv ID should replace the cluster ID in this context.  The controller will also use this information to create the Agent CRs in the same namespaces as their associated InfraEnvs.
 
-- From the above point, an ISO will now include the image ID and an optional cluster ID.  These will be passed in the host registration API.
-
-- Internally, in the SQL DB, hosts currently have two primary keys - the host ID and the cluster ID.  This will continue to be the case to avoid a difficult migration and rewrite.  We will create a new type of cluster that represents a tenant.  Agents not associated to a "real" cluster will be associated with the tenant cluster.  We will continue to have separate host entities for each cluster/tenant association.  A host may be bound to different clusters at different times, but will always be associated only with the image it booted from.  Thus status, events, logs, and other host information will always be in the context of a cluster, as today.
+- Internally, in the SQL DB, hosts currently have two primary keys - the host ID and the cluster ID.  Now the primary keys will be host ID and InfraEnv ID.
 
 - Currently the agent writes the image from the ISO to disk, which means all Agents that form a cluster need to be running with the same version.  Instead, the agent should download the correct version for the cluster being formed if it doesn't match what's in the ISO.
 
 - When moving a host from a source to dest cluster the service needs to make sure that the source cluster is not installing (can be before or after) and that the dest cluster is before installation.  If the source cluster is before installation, the host will automatically go to disconnected state, which is the correct behavior.  However, the service should update its state manually so it doesn't take 3+ minutes (e.g., by resetting the timestamp of when it last checked in).  If the host doesn't exist in the dest cluster a new host will be created (register_host).  If it does exist, we reset its state and properties but not its events.
 
 - There should be one pull secret for the ISO to pull the necessary images (Infrastructure Admin's'), and another pull secret in the cluster definition for the installed cluster to use (Cluster Creator's).
 
+- Token revocation currently relies on the cluster ID.  It should now rely on the InfraEnv ID.
```

Comment:
Didn't we say that it will be hard to revoke a token based on that since we don't want to delete InfraEnv?

Reply:
Previously I was considering that they would need to delete the tenant/namespace (old model).  Deleting an InfraEnv is a bit painful but not the end of the world in the hopefully rare case of invalidating a token.